src/main/java/com/xjeffrose/chicago/client/BaseChicagoClient.java

```diff
@@ -55,14 +55,17 @@ public void nodeRemoved() {
   protected EventLoopGroup evg;
 
 
-  public BaseChicagoClient(String address){
+  public BaseChicagoClient(String address) throws InterruptedException{
     this.single_server = true;
-    this.zkClient = null;
+    // this is only for test purpose. The zkServer should run on the same machine as that of the chicago server.
+    this.zkClient = new ZkClient(address.split(":")[0]+":2181", false);
+    this.zkClient.start();
     this.quorum = 1;
     ArrayList<String> nodeList = new ArrayList<>();
     nodeList.add(address);
     this.rendezvousHash =  new RendezvousHash(Funnels.stringFunnel(Charset.defaultCharset()), nodeList, quorum);
-    clientNodeWatcher = null;
+    // only if the zkserver runs in the same machine as the chicago server [Test purpose]
+    clientNodeWatcher = new ClientNodeWatcher(zkClient, rendezvousHash, listener);
```

Comment:
I assumed that this constructor would be more or less used for development with ZK running on the same machine as that of the "address" variable. Let me know if this is ok. I'll change this back. But this was needed for my local testing purpose.